consumer.go

```diff
@@ -102,6 +102,7 @@ func (c *Consumer) consumeJobIter(iter queue.JobIter) error {
 		}
 
 		if err != nil {
+			iter.Close()
```

Comment:
are we ok not checking if closing the `iter` returns an error?

Reply:
Now that error is logged.